lib/component.js

```diff
@@ -650,7 +664,7 @@ class Component extends WebComponent {
 
     const hook = (this.getConfig(`hooks`) || {})[hookName];
     if (hook) {
-      hook(...params);
+      hook.apply(this, params);
```

Comment:
just make sure the hook has `this` correctly binded

Reply:
I'm inclined to avoid changing this as a drive-by change. At this point it's long-standing behavior that someone may rely upon, and it's also not especially necessary. If people want `this` binded, they can easily use arrow functions.

Reply:
hmm a quick search in the codebase all hooks are using arrow function. i could revert it back to see if my testcase could pass

Reply:
This is technically a public project/package. I don't think we should break long-standing behavior unless necessary. I'd recommend you revert it back.

Reply:
https://github.com/mixpanel/panel/pull/161/files#diff-094bb12e1d2490b9c34b42b64c7e2f9fb0b84fd96d4f42eb2cee4e31296453adR85 the challenge is there's a testcase targeting the hook `this` context, i can't get it work without tweaking this apply thing

Reply:
You should be able to reframe that test code without `this` binding:
```typescript
    child._config.hooks = {
      preUpdate(stateUpdate, paramsUpdate) {
        expect(child.params.str).to.equal(``);
        expect(paramsUpdate.str).to.equal(`abc`);
      },
    };
```

Reply:
ah I was trapped in my own little mind maze, thx for the help 🙈